### PR TITLE
Correctly calculate used space for quota with external storage

### PR DESCRIPTION
**Pull request by Ardinis**
issue #4348

### CODE REVIEW

File: lib/private/legacy/helper.php [hunk 1]
```diff
@@ -537,7 +537,7 @@ public static function getStorageInfo($path, $rootInfo = null) {
 		$includeExtStorage = \OC::$server->getSystemConfig()->getValue('quota_include_external_storage', false);
 
 		if (!$rootInfo) {
-			$rootInfo = \OC\Files\Filesystem::getFileInfo($path, false);
+			$rootInfo = \OC\Files\Filesystem::getFileInfo($path, $includeExtStorage);
```
Comment (icewind1991) [inline lib/private/legacy/helper.php]: This should be `'ext'`, when using `true` it also includes incoming shares

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: lib/private/legacy/helper.php (hunk 1 above)
Comment (Ardinis) [inline lib/private/legacy/helper.php]: I am not sure of what you are meaning here.
By using $includeExtStorage external storage are counted in quota calculation for the used part.

Comment (icewind1991) [inline lib/private/legacy/helper.php]: You can pass `'ext'` as the seccond argument of `getFileInfo` to have it include external storages but not shares.

When passing `true` it includes both shares and external storages

File: lib/private/legacy/helper.php [hunk 2]
```diff
@@ -537,7 +537,7 @@ public static function getStorageInfo($path, $rootInfo = null) {
 		$includeExtStorage = \OC::$server->getSystemConfig()->getValue('quota_include_external_storage', false);
 
 		if (!$rootInfo) {
-			$rootInfo = \OC\Files\Filesystem::getFileInfo($path, false);
+			$rootInfo = \OC\Files\Filesystem::getFileInfo($path, 'ext');
```
Comment (icewind1991) [inline lib/private/legacy/helper.php]: It should still only apply external storages when the config is set.

```php
$rootInfo = \OC\Files\Filesystem::getFileInfo($path, $includeExtStorage ? 'ext' : false);
```

**Review by icewind1991 — CHANGES_REQUESTED** [review]

**Comment by rullzer** [comment]
@Ardinis could you address the comments of @icewind1991 ?

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Could you also have a look at the failing unit test:

```
1) Test\HelperStorageTest::testGetStorageInfoIncludingExtStorageWithNoUserQuota
Failed asserting that 22 matches expected 5.
/drone/src/github.com/nextcloud/server/tests/lib/HelperStorageTest.php:162
```

And could you squash all your commits into one, because 3 commits is a bit much for a 1 line change :wink:

And if this commit then also would have a sign-off I would be more than happy :) https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work

**Comment by MorrisJobke** [comment]
I fixed the unit tests: #4816

**Pull request closed without merging**